torch/csrc/jit/passes/frozen_graph_optimizations.cpp

```diff
@@ -22,6 +23,7 @@ void OptimizeFrozenGraph(
       FoldFrozenConvMulOrDiv(graph);
     }
   }
+  DecomposeLinearToMatmulAdd(graph);
```

Comment:
Hmmm, doing this on CPU is going to interfere with our mkldnn fusion pass which uses mkldnn's implementation of linear. Maybe in this first PR we not yet enable it or make sure that we are only rewriting `linear` nodes on cuda ?